controllers/releaseplan/controller.go

```diff
@@ -60,15 +63,26 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	return controller.ReconcileHandler([]controller.Operation{
 		adapter.EnsureOwnerReferenceIsSet,
+		adapter.EnsureMatchingInformationIsSet,
 	})
 }
 
 // Register registers the controller with the passed manager and log.
 func (c *Controller) Register(mgr ctrl.Manager, log *logr.Logger, _ cluster.Cluster) error {
 	c.client = mgr.GetClient()
-	c.log = log.WithName("releasePlan")
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.ReleasePlan{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		For(&v1alpha1.ReleasePlan{}, builder.WithPredicates(predicate.GenerationChangedPredicate{}, predicates.ReleasePlanMatchUpdatePredicate())).
+		Watches(&v1alpha1.ReleasePlanAdmission{}, &handlers.EnqueueForMatchingUpdate{},
+			builder.WithPredicates(predicates.ReleasePlanAdmissionMatchUpdatePredicate())).
 		Complete(c)
 }
+
+// SetupCache indexes fields for each of the resources used in the releaseplan adapter in those cases where filtering by
+// field is required.
+// NOTE: Both the release and releaseplan controller need this ReleasePlanAdmission cache. However, it only needs to be added
```

Comment:
@davidmogar what are your thoughts on this?

Reply:
To check what happens when it's added two times. If nothing gets duplicated, I would uncomment this.

Reply:
Controller fails to start when duplicated

Reply:
This has not changed. As mentioned, declare the SetupCache but leave it empty with a comment pointing to the ReleasePlanAdmissions cache (or viceversa) and explaining what you explain in the note.